core/include/scipp/core/element/reduction.h

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+
+#include "scipp/common/overloaded.h"
+#include "scipp/core/dtype.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/except.h"
+#include "scipp/units/unit.h"
+
+namespace scipp::core::element {
+
+template <class... Ts>
+constexpr arg_list_t<std::tuple<event_list<Ts>, event_list<Ts>, bool>...>
+    flatten_arg_list{};
+
+constexpr auto flatten = overloaded{
```

Comment:
Any reason why sum would not also be here?

Reply:
I think `sum` does not exist? Wasn't it just using `+=`?

Reply:
Ah ok. I had assumed it would be there, based on the higher-level, https://scipp.github.io/generated/scipp.sum.html but you're right.